tests/scalers_go/gcp_pubsub/gcp_pubsub_test.go

```diff
@@ -236,42 +240,64 @@ func getTemplateData() (templateData, templateValues) {
 	base64GcpCreds := base64.StdEncoding.EncodeToString([]byte(gcpKey))
 
 	return templateData{
-			TestNamespace:    testNamespace,
-			SecretName:       secretName,
-			GcpCreds:         base64GcpCreds,
-			DeploymentName:   deploymentName,
-			ScaledObjectName: scaledObjectName,
-			SubscriptionID:   subscriptionID,
-			SubscriptionName: subscriptionName,
-			MaxReplicaCount:  maxReplicaCount,
+			TestNamespace:       testNamespace,
+			SecretName:          secretName,
+			GcpCreds:            base64GcpCreds,
+			DeploymentName:      deploymentName,
+			ScaledObjectName:    scaledObjectName,
+			SubscriptionID:      subscriptionID,
+			SubscriptionName:    subscriptionName,
+			MaxReplicaCount:     maxReplicaCount,
+			ActivationThreshold: activationThreshold,
 		}, templateValues{
 			"secretTemplate":       secretTemplate,
 			"deploymentTemplate":   deploymentTemplate,
 			"scaledObjectTemplate": scaledObjectTemplate,
 			"gcpSdkTemplate":       gcpSdkTemplate}
 }
 
+func publishMessages(t *testing.T, count int) {
+	t.Log(fmt.Sprintf("--- publishing %d messages ---", count))
+	publish := fmt.Sprintf(
+		"%s/bin/bash -c -- 'for i in {1..%d}; do gcloud pubsub topics publish %s --message=AAAAAAAAAA;done'",
+		gsPrefix,
+		count,
+		topicID)
+	_, err := ExecuteCommand(publish)
+	assert.NoErrorf(t, err, "cannot publish messages to pubsub topic - %s", err)
+}
+
+func testDontScaleUpIfBelowThreshold(t *testing.T, kc *kubernetes.Clientset) {
+	t.Log("--- testing not scaling if below threshold ---")
+
+	publishMessages(t, activationThreshold)
+
+	t.Log("--- waiting to see replicas are not scaled up ---")
+
+	for i := 0; i < 24; i++ {
+		ok := WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, 0, 1, 1)
+		assert.True(t, ok, "replica count should be 0 if below activation threshold")
+		if !ok {
+			return
+		}
+
+		time.Sleep(time.Duration(10) * time.Second)
+	}
```

Comment:
There is already a helper for doing this `AssertReplicaCountNotChangeDuringTimePeriod`. You can see an example [here](https://github.com/kedacore/keda/blob/main/tests/scalers_go/aws_dynamodb/aws_dynamodb_test.go#L171)